AlphaWallet/Transactions/EtherscanSingleChainTransactionProvider.swift

```diff
@@ -290,13 +290,13 @@ class EtherscanSingleChainTransactionProvider: SingleChainTransactionProvider {
 
             firstly {
                 EtherscanSingleChainTransactionProvider.functional.fetchTransactions(startBlock: startBlock, sortOrder: sortOrder, session: coordinator.session, alphaWalletProvider: coordinator.alphaWalletProvider, tokensDataStore: coordinator.tokensDataStore, queue: coordinator.queue)
-            }.done { [weak self] transactions in
+            }.done(on: queue, { [weak self] transactions in
                 guard let strongSelf = self else { return }
                 guard !strongSelf.isCancelled else { return }
                 coordinator.addOrUpdate(transactions: transactions)
-            }.catch { e in
+            }).catch(on: queue, { e in
                 error(value: e, rpcServer: coordinator.session.server, address: self.session.account.address)
-            }.finally { [weak self] in
+            }).finally(on: queue, { [weak self] in
```

Comment:
Maybe this should be back on main?

Reply:
since we made ability to update reqlm on synced background queue, we are also able to add objects to realm, on background threads

Reply:
Hmm.. wouldn't that be hard to keep track of and introduce bugs? For example line 296:

```
coordinator.addOrUpdate(transactions: transactions)
```

So the body of this function `addOrUpdate(transactions:) and everything it calls might run in the background.